### PR TITLE
Maker: Inject board into components

### PR DESCRIPTION
**Pull request by islemaster**
Prepares playground component initialization for a new approach to run/reset. This PR should _not_ result in a behavior change.

_Pair: @bcjordan_

Johnny-Five components can be initialized with or without injecting a `board`.  Without one (before this PR) they automatically attach to the first board object found in the page's static cache.  Now we explicitly provide a board for the component to use, breaking the static dependency. For details see [`Board.Component`](https://github.com/rwaldron/johnny-five/blob/master/lib/board.js#L775-L802) (called as part of every component's constructor) and [`Board.mount`](https://github.com/rwaldron/johnny-five/blob/master/lib/board.js#L730-L771) (badly named?) within johnny-five.

In the process we've broken up the large `initializeCircuitPlaygroundComponents` method into lots of small methods for initializing individual components, and brought those methods under test.  We are actually testing against real johnny-five and playground-io.  We mock firmata-js (playground-io's superclass, and effectively our 'hardware interface' at this layer) by swapping in [rwaldron/mock-firmata](https://github.com/rwaldron/mock-firmata) for firmata imports via Webpack configuration.  This was a real headache to figure out so there's a good bit of commenting inline - please let me know if you think more is needed.

### CODE REVIEW

File: apps/package.json [hunk 1]
```diff
@@ -133,6 +133,7 @@
     "mocha": "^3.1.2",
     "mochify": "2.18.0",
     "mockfirebase": "^0.12.0",
+    "mock-firmata": "0.1.0",
```
Comment (islemaster) [inline apps/package.json:136]: Rick Waldron's [mock-firmata](https://github.com/rwaldron/mock-firmata), which gets [used in the johnny-five unit tests](https://github.com/rwaldron/johnny-five/blob/master/test/common/bootstrap.js#L96).

File: apps/test/unit/lib/kits/maker/PlaygroundComponentsTest.js [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+      // 20 sysex calls during initialization, two per LED.
+      for (let i = 0; i < 10; i++) {
+        // TODO (bbuchanan): Record what these calls mean.
+        expect(board.io.sysexCommand.getCall(i*2)).to.have.been.calledWith([0x40, 0x10, i, 0, 0, 0, 0]);
+        expect(board.io.sysexCommand.getCall(i*2+1)).to.have.been.calledWith([0x40, 0x11]);
+      }
+    });
+
+    it('can initialize a second set of leds with a second board', () => {
```
Comment (islemaster) [inline apps/test/unit/lib/kits/maker/PlaygroundComponentsTest.js:49]: This test in particular is making sure that the components in question attach to the board we passed in, not just to the first board created, which will be important for reset purposes.

File: apps/test/unit/lib/kits/maker/PlaygroundComponentsTest.js [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+      expect(sensor).to.be.an.instanceOf(five.Sensor);
+      // Doesn't use sysex at first
+      expect(board.io.sysexCommand.callCount).to.equal(0);
+    });
+
+    it('gets sensor methods', () => {
+      const sensor = initializeSoundSensor(board);
+      expect(sensor).to.haveOwnProperty('start');
+      expect(sensor).to.haveOwnProperty('getAveragedValue');
+      expect(sensor).to.haveOwnProperty('setScale');
```
Comment (Hamms) [inline apps/test/unit/lib/kits/maker/PlaygroundComponentsTest.js:75]: I'm not totally convinced of the value of these; what are they saying that `expect(sensor).to.be.an.instanceOf(five.Sensor);` isn't?

Comment (islemaster) [reply apps/test/unit/lib/kits/maker/PlaygroundComponentsTest.js:75]: These particular methods are not part of `five.Sensor` - we monkeypatch them on with [`addSensorFeatures()`](https://github.com/code-dot-org/code-dot-org/blob/e4720aaa135a10fa5b8c9d18d65de913495a884b/apps/src/lib/kits/maker/PlaygroundComponents.js#L107) to create the API our curriculum team requested.

File: apps/test/unit/lib/kits/maker/PlaygroundComponentsTest.js [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+      expect(board.io.sysexCommand.getCall(5)).to.have.been.calledWith([0x40, 0x41, 9]);
+      expect(board.io.sysexCommand.getCall(6)).to.have.been.calledWith([0x40, 0x41, 10]);
+      expect(board.io.sysexCommand.getCall(7)).to.have.been.calledWith([0x40, 0x41, 12]);
+    });
+  });
+});
+
+/**
+ * Generate a test board object.
+ * From rwaldrong/johnny-five's newBoard() test helper:
```
Comment (bcjordan) [inline apps/test/unit/lib/kits/maker/PlaygroundComponentsTest.js:161]: s/rwaldrong/rwaldron

**Pull request merged**